src/lib/controller.js

```diff
@@ -0,0 +1,85 @@
+import {store} from "./store";
+import {updateUrl} from "./fb";
+import {runLighthouse, fetchReports} from "./lighthouse-service";
+import serial from "./serial";
+
+const serialLighthouseTask = serial(store);
+
+export const requestRunLighthouse = store.action(async (state, url) => {
+  if (state.activeLighthouseUrl) {
+    return null;
+  }
+
+  // We're writing the user's new URL choice to local state, but also kicking
+  // off a Lighthouse run here. We store the active URL running through
+  // Lighthouse as their preferred URL could change because of another
+  // browser, and this prevents results being appropritioned to the wrong URL.
+  store.setState({
+    userUrl: url,
+    activeLighthouseUrl: url,
+    lighthouseError: null,
+  });
+
+  return serialLighthouseTask(async (state) => {
```

Comment:
Another possibility is compare the LH result with whatever the local user url is before displaying the results. If there's a mismatch just throw an error. That might let us remove some of the serialization code to simplify things a bit.